### PR TITLE
Integrate with Transifex

### PR DESCRIPTION
**Pull request by sbrunner**
I backport from Lux:
- The state manager
- The `switchLanguage` and the `initLanguage` methods
and adapt them for gmf/ngeo ...

TODO:
 * [x] Copy the json to github pages

fix: #415

### CODE REVIEW

File: docs/guidelines.md [hunk 1]
```diff
@@ -496,9 +496,11 @@ in `ngeo`.
 
 ### template vs templateUrl
 
-There is at the moment no real exigence concerning this point. It's up to you to
-determine if the template is simple enough to be written inline in the directive
-code with the`template` attribute.
+The only technical reason to use `templateUrl` instance of `template` is that the
```
Comment (fgravin) [inline docs/guidelines.md]: type instead

File: docs/guidelines.md [hunk 2]
```diff
@@ -496,9 +496,11 @@ in `ngeo`.
...
+template don't support i18n.
```
Comment (fgravin) [inline docs/guidelines.md]: doesn't

File: docs/guidelines.md (hunk 1 above)
Comment (sbrunner) [inline docs/guidelines.md]: ? I don't understand ...

Comment (fgravin) [inline docs/guidelines.md]: `templateUrl` instance of `template` it's `instead` no ?

Comment (sbrunner) [inline docs/guidelines.md]: Effectively yes, thanks :-)

**Comment by fgravin** [comment]
Could you please explain in few words the mechanism ?
The new make targets, what does it requires to add in our code to use it etc...

Thanks

**Comment by sbrunner** [comment]
* Collect i18n strings from the sources files with https://github.com/camptocamp/ngeo/pull/477/files#diff-0aeb17f9bf66aae856c7fd4513b9852c
* On a commit on master Travis will send it to Transifex
* Gets the translations back from Transifex
* Build a JSON with https://github.com/camptocamp/ngeo/pull/477/files#diff-eae9dd470245630a5d683f7986c2fcb1
* Use it in https://github.com/camptocamp/ngeo/pull/477/files#diff-15f5699aebde24ce04cd82d2cc47452dR217

It's in place only for gmf because we haven't any string to translate in ngeo...

**Comment by sbrunner** [comment]
Result: http://camptocamp.github.io/ngeo/transifex/examples/contribs/gmf/apps/mobile/index.html?lang=fr

File: Makefile [hunk 3]
```diff
@@ -474,6 +484,57 @@ contribs/gmf/build/mobile.css: $(GMF_APPS_MOBILE_LESS_FILES) \
...
+	echo "token =" >> $@
+
+.tx/config: .tx/config.mako .build/python-venv/bin/mako-render
+	PYTHONIOENCODING=UTF-8 .build/python-venv/bin/mako-render \
+		--var "git_branch=$(TX_GIT_BRANCH)" $< > $@
+
+#.build/locale/ngeo.pot: lingua.cfg .build/node_modules.timestamp \
+#		$(NGEO_DIRECTIVES_PARTIALS_FILES) $(NGEO_JS_FILES)
+#	mkdir -p $(dir $@)
+#	node buildtools/extract-messages.js $(NGEO_DIRECTIVES_PARTIALS_FILES) $(NGEO_JS_FILES) > $@
```
Comment (ger-benjamin) [inline Makefile]: These comments can be removed ?

File: Makefile [hunk 4]
```diff
@@ -474,6 +484,57 @@ contribs/gmf/build/mobile.css: $(GMF_APPS_MOBILE_LESS_FILES) \
...
+	.build/python-venv/bin/pip install transifex-client
+	touch $@
+
+.PHONY: transifex-send
+transifex-send: .tx/config .build/python-venv/bin/tx \
+		.build/locale/gmf.pot
+		# .build/locale/ngeo.pot
+	.build/python-venv/bin/tx push --source
+
+#.build/locale/%/LC_MESSAGES/ngeo.po: .tx/config .build/python-venv/bin/tx
```
Comment (ger-benjamin) [inline Makefile]: These comments can be removed ?

Comment (sbrunner) [inline Makefile]: I add those blocks to be almost ready when we will have something to translate in ngeo ...

**Comment by sbrunner** [comment]
Rebased and up ...

File: .travis.yml [hunk 5]
```diff
@@ -23,4 +23,6 @@ after_success:
 - export GIT_BRANCH=${TRAVIS_BRANCH}
 - git config --global user.name "Travis"
 - git config --global user.email "travis@travis-ci.org"
-- '[ "${TRAVIS_PULL_REQUEST}" = "false" ] && make gh-pages'
```
Comment (fgravin) [inline .travis.yml]: Done in https://github.com/camptocamp/ngeo/pull/506/files ?

Comment (sbrunner) [inline .travis.yml]: no, we add an other file ...

**Comment by fgravin** [comment]
It is hard to review but it globally looks good.
Thanks, you can merge if you have the expected behavior when testing your banch.

**Comment by sbrunner** [comment]
Effectively I understand, thanks @fgravin :-)

**Pull request merged**